app/ante/vesting.go

```diff
@@ -80,11 +90,34 @@ func (vtd EthVestingTransactionDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx,
 			)
 		}
 
-		// Error if account has locked coins (before surpassing all lockup periods)
-		islocked := clawbackAccount.HasLockedCoins(ctx.BlockTime())
-		if islocked {
-			return ctx, errorsmod.Wrapf(vestingtypes.ErrVestingLockup,
-				"cannot perform Ethereum tx with clawback vesting account, that has locked coins: %s", vested,
+		// Check to make sure that the account does not exceed its spendable balances.
+		// This transaction would fail in processing, so we should prevent it from
+		// moving past the AnteHandler.
+		msgValue := sdk.NewCoin(evmDenom, math.NewIntFromBigInt(msgEthTx.AsTransaction().Value()))
+		address := acc.GetAddress()
+
+		// Since there can be multiple transactions from different accounts, we track each account's total
+		// spend to compare against its unlocked balances.
+		totalSpend, ok := totalSpendByAddress[address.String()]
+		if !ok {
+			totalSpend = msgValue
+		} else {
+			totalSpend = totalSpend.Add(msgValue)
+		}
+		totalSpendByAddress[address.String()] = totalSpend
+
+		// Check that the clawbackAccount has sufficient unlocked tokens to cover all requested spending.
+		// lockedBalance defaults to zero if not found.
+		_, lockedBalance := clawbackAccount.LockedCoins(ctx.BlockTime()).Find(evmDenom)
+
+		spendableBalance, err := vtd.bk.GetBalance(ctx, address, evmDenom).SafeSub(lockedBalance)
```

Comment:
you can move the GetBalance check to the top to short-circuit in case the user has no balance